lib/marc_cleanup/record_level.rb

```diff
@@ -13,16 +15,55 @@ def bad_utf8?(record)
     record.to_s.scrub != record.to_s
   end
 
+  def bad_utf8_scrub_datafield(record:, field:, field_index:)
+    field.subfields.each_with_index do |subfield, subfield_index|
+      new_value = bad_utf8_scrub_value(subfield.value)
+      record.fields[field_index].subfields[subfield_index].value = new_value
+    end
```

Comment:
Would this also do the trick, without the need to track the subfield index?

```suggestion
    field.subfields.each do |subfield|
      subfield.value = bad_utf8_scrub_value(subfield.value)
    end
```

Reply:
Modifying the value in place doesn't seem to work. However, by making bad_utf8_scrub_datafield return a field, I think the code will be cleaner.